core/templates/pages/topic-editor-page/navbar/topic-editor-navbar.component.ts

```diff
@@ -125,7 +128,28 @@ angular.module('oppia').component('topicEditorNavbar', {
         return UndoRedoService.getChangeCount();
       };
 
+      $scope.updateTopicEditorBrowserTabsUnsavedChangesStatus = function() {
+        var topic = TopicEditorStateService.getTopic();
+
+        const topicEditorBrowserTabsInfo:
+          EntityEditorBrowserTabsInfo = (
+            LocalStorageService.getEntityEditorBrowserTabsInfo(
+              EntityEditorBrowserTabsInfoDomainConstants
+                .OPENED_TOPIC_EDITOR_BROWSER_TABS, topic.getId()));
+
+        if (topicEditorBrowserTabsInfo && $scope.getChangeListLength() > 0 &&
+              !topicEditorBrowserTabsInfo.doesSomeTabHaveUnsavedChanges()
+        ) {
+          topicEditorBrowserTabsInfo.setSomeTabHasUnsavedChanges(true);
+          LocalStorageService.updateEntityEditorBrowserTabsInfo(
+            topicEditorBrowserTabsInfo,
+            EntityEditorBrowserTabsInfoDomainConstants
+              .OPENED_TOPIC_EDITOR_BROWSER_TABS);
+        }
+      };
+
       $scope.isTopicSaveable = function() {
+        $scope.updateTopicEditorBrowserTabsUnsavedChangesStatus();
```

Comment:
Why are we calling this here? How often will it be executed?

Reply:
Same question. Is the function resource intensive or does it finish quickly?

Reply:
This function will update local storage only if the value of `doesSomeTabHaveUnsavedChanges` is false in the local storage and there are unsaved changes in the correct tab. Hence, between two save operations, this function will update the local storage once only.

Reply:
If you try adding a console log here, how many times does it print to the console?

Reply:
Update: I have optimised the process and shifted this function to `topic update service now`. The results are as follows:

** Before (It was running every time `isTopicSaveable` was running i.e. on each keystroke to a particular field) **

https://user-images.githubusercontent.com/70739349/167247517-b617e425-4d16-4e08-ba85-2cbd1c9784dd.mp4

** After (It is running once while applying the draft changes onto the topic)

https://user-images.githubusercontent.com/70739349/167247561-25ea2d38-9f78-4788-81cf-0412184cc344.mp4